fabric-screen-handler-api-v1/src/main/java/net/fabricmc/fabric/mixin/screenhandler/ServerPlayerEntityMixin.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.screenhandler;
+
+import java.util.OptionalInt;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.network.Packet;
+import net.minecraft.screen.NamedScreenHandlerFactory;
+import net.minecraft.screen.ScreenHandler;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.api.screenhandler.v1.ExtendedScreenHandlerFactory;
+import net.fabricmc.fabric.impl.screenhandler.ExtendedScreenHandlerType;
+import net.fabricmc.fabric.impl.screenhandler.ScreenHandlersImpl;
+
+@Mixin(ServerPlayerEntity.class)
+public class ServerPlayerEntityMixin {
+	@Shadow
+	private int screenHandlerSyncId;
+
+	@Unique
+	private final ThreadLocal<ScreenHandler> fabric_openedScreenHandler = new ThreadLocal<>();
+
+	@Inject(method = "openHandledScreen(Lnet/minecraft/screen/NamedScreenHandlerFactory;)Ljava/util/OptionalInt;", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/network/ServerPlayNetworkHandler;sendPacket(Lnet/minecraft/network/Packet;)V"), locals = LocalCapture.CAPTURE_FAILHARD)
+	private void fabric_onOpenHandledScreen_cacheScreenHandler(NamedScreenHandlerFactory factory, CallbackInfoReturnable<OptionalInt> info, ScreenHandler handler) {
```

Comment:
Are we going for prefixing the inject methods with `fabric` and `_`? I am seeing mostly no prefix _on inject_, this is an inject so conflicting names shouldn't matter right?

Reply:
The fabric_ prefix is also for exception stacktraces because `VanillaClass.someMethod$onSomeMethod` doesn't tell what mod caused the error, unlike `fabric_onSomeMethod`.

Reply:
We might want to have a standard in place for this, 99% of mixins have no prefix right now, we might want to refactor all of them if that's what we are going for.

Reply:
But that's a topic for another day, let's not bloat this thing

Reply:
That is actually an issue I brought up already https://github.com/FabricMC/fabric/issues/510

Reply:
Prefixes are not necessary on `@Inject` methods, since Mixin prepends special signature info onto each method. For `MixinAbstractFurnaceBlockEntity`'s [`fuelTimeMapHook`](https://github.com/FabricMC/fabric/blob/1.16/fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/mixin/content/registry/MixinAbstractFurnaceBlockEntity.java#L34) method, Mixin names it `handler$zeh000$fuelTimeMapHook`. It's also given a `@MixinMerged` annotation noting the mixin class that added the injection. If the `fabric_` prefix here is because Mixin error handling doesn't expose the cause well enough, then we can just improve error handling on our Mixin fork. This method should just be named `cacheScreenHandler`.

Reply:
we should use prefix, though i personally prefer `fabric$` prefix as `_` looks too ugly

Reply:
We should follow the standard in the rest of Fabric API or create a new PR to enforce the `fabric$` prefix standard across the rest of the API. Renaming mixin methods is not breaking, so it can be done safely at any time. Stick with the API convention for now.

Reply:
There's no API convention, though. Fabric uses all of those different styles that i5 listed in #510.

Reply:
oh god, really? I could have sworn there was one already

Reply:
I searched through some mixins and 99% of them have no prefix, the example in i5's issue for `fabric_` is not an inject, most other `fabric_` are overrides, the current majority is no prefix.

I am not saying that no prefix is the best style but mostly injects in fabric API have no prefixes right now, and we should stick to the masses until we have a PR to change this.